adoptopenjdk-api-v3-models/src/main/kotlin/net/adoptopenjdk/api/v3/models/Variant.kt

```diff
@@ -28,6 +29,8 @@ class Variant {
         version: Int,
         @JsonProperty("lts")
         lts: Boolean?,
+        @JsonProperty("obsoleteRelease")
+        obsoleteRelease: Boolean,
```

Comment:
In the PR conversation there has been mention of `eol_reached` by @tushev so just wondering if `obsoleteRelease` is the right name for this new field on the variant - anyway being consistent across the API is important whatever we decide.

Reply:
Whichever one we choose depends on what we want the attribute to imply. The way I parsed the two names were as follows:
- `eol_reached` = End of life reached. That implies the end of life of the JDK has been reached but may still be built at Adopt (due to delay's in disabling it, need to continue building it for specific reasons, etc).
- `obsolete_release` = The release has been declared obsolete by Adopt. That implies the end of life of the JDK has no necessarily been reached but is no longer built at Adopt for whatever reason.
Could use some opinions on both sides for this? @tushev @karianna @johnoliver

Reply:
I am fine with the obsolete_release term

Reply:
@M-Davies ,

What my updater requires, is a some kind of `"warn_about_this_release": "true"` field, that means that the updater should warn the user **that them should not be using this release any more**.

1. As far as understand, querying something like https://api.adoptopenjdk.net/v3/assets/latest/6/openj9 and getting`[]` in response means that this release has binaries withdrawn and the user should be moving away from it. 

**BTW, could `[]` response be a result of an internal error?** (If yes, than I should perform an additional request to [v3/info/available_releases](https://api.adoptopenjdk.net/v3/info/available_releases) to make sure the release is obsolete).

2. At the same time, there could be a situation when the release is still offered at Adopt(ium), but the users should move from it anyway. 

So, **from this point of view, there should be** `warn_about_this_release = eol_reached || obsolete_release` **field in API** .

Alternatively, there could only some other logic. I prefer not to do any decisions on the updater's side, and have just a single boolean field that will command the updater (and whose logic could be changed in the future if necessary).
<hr>
So, (#206)


* I need a kind of `releases_to_warn_about` array in [v3/info/available_releases](https://api.adoptopenjdk.net/v3/info/available_releases) which contains a list of ALL releases that the users should stop using
* I need `warn_about_this_release` boolean field in [v3/assets/latest/{feature_version}/{jvm_impl}](https://api.adoptopenjdk.net/v3/assets/latest/8/openj9)->item. This field saves extra request(s), thus allowing for faster check. Not really top priority.

Maybe, let's just introduce `obsolete_releases` as a beginning, and introduce `warn_about_this_release = eol_reached || obsolete_release` later (if that will be ever necessary).